















<!doctype html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta http-equiv="last-modified" content="2018-10-20 22:53:04 +1100">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <!-- meta "search-domain" used for google site search function google_search() -->
    <meta name="search-domain" value="">
    <link rel="stylesheet" type="text/css" href="../assets/css/bootstrap.css" />
    <link rel="stylesheet" type="text/css" href="../assets/css/bootstrap-theme.css" />
    <link rel="stylesheet" type="text/css" href="../assets/css/lesson.css" />
    <link rel="stylesheet" type="text/css" href="../assets/css/syntax.css" />


    <!-- HTML5 shim and Respond.js for IE8 support of HTML5 elements and media queries -->
    <!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
    <!--[if lt IE 9]>
	<script src="https://oss.maxcdn.com/html5shiv/3.7.2/html5shiv.min.js"></script>
	<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
	<![endif]-->
    <title>CBMS Advanced Biomolecular Analysis - Genomics: Assembly Mapping and Binning</title>
  </head>
  <body>

    







    <div class="container">
      
<nav class="navbar navbar-default">
  <div class="container-fluid">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>



      
      <a class="navbar-brand" href="../index.html">Home</a>

    </div>
    <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
      <ul class="nav navbar-nav">


        
	
        <li><a href="../setup.html">Setup</a></li>

        
        <li class="dropdown">
          <a href="../" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">Episodes <span class="caret"></span></a>
          <ul class="dropdown-menu">
            
            <li><a href="../01-intro/index.html">Introducing the Shell</a></li>
            
            <li><a href="../02-filedir/index.html">Navigating Files and Directories</a></li>
            
            <li><a href="../03-create/index.html">Working With Files and Directories</a></li>
            
            <li><a href="../04-pipefilter/index.html">Pipes and Filters</a></li>
            
            <li><a href="../06-Genome_assembly/index.html">Genome Assembly Workflow</a></li>
            
            <li><a href="../06-trimming/index.html">Trimming and Filtering</a></li>
            
            <li><a href="../07-meta-genome_assembly/index.html">Genome Assembly Workflow - Week 2</a></li>
            
            <li><a href="../08-assembly_mapping_and%20binning/index.html">Assembly Mapping and Binning</a></li>
            
            <li><a href="../09-annotation-of-genome-bins/index.html">Annotation of genome bins. Week 2-3</a></li>
            
            <li><a href="../12-taxonomic-assignment/index.html">Taxonomic assignment using LCA. Week 3-4</a></li>
            
	    <li role="separator" class="divider"></li>
            <li><a href="../aio.html">All in one page (Beta)</a></li>
          </ul>
        </li>
	

	
	
        <li class="dropdown">
          <a href="../" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">Extras <span class="caret"></span></a>
          <ul class="dropdown-menu">
            <li><a href="../reference.html">Reference</a></li>
            
            <li><a href="../about/index.html">About</a></li>
            
            <li><a href="../discuss/index.html">Discussion</a></li>
            
            <li><a href="../figures/index.html">Figures</a></li>
            
            <li><a href="../guide/index.html">Instructor Notes</a></li>
            
          </ul>
        </li>
	

	
        <li><a href="../LICENSE.html">License</a></li>
	
	<li><a href="/edit/gh-pages/_episodes/08-assembly_mapping_and binning.md">Improve this page <span class="glyphicon glyphicon-pencil" aria-hidden="true"></span></a></li>
	
      </ul>
      <form class="navbar-form navbar-right" role="search" id="search" onsubmit="google_search(); return false;">
        <div class="form-group">
          <input type="text" id="google-search" placeholder="Search..." aria-label="Google site search">
        </div>
      </form>
    </div>
  </div>
</nav>

















<div class="row">
  <div class="col-xs-1">
    <h3 class="text-left">
      
      <a href="../07-meta-genome_assembly/index.html"><span class="glyphicon glyphicon-menu-left" aria-hidden="true"></span><span class="sr-only">previous episode</span></a>
      
    </h3>
  </div>
  <div class="col-xs-10">
    
    <h3 class="maintitle"><a href="../">CBMS Advanced Biomolecular Analysis - Genomics</a></h3>
    
  </div>
  <div class="col-xs-1">
    <h3 class="text-right">
      
      <a href="../09-annotation-of-genome-bins/index.html"><span class="glyphicon glyphicon-menu-right" aria-hidden="true"></span><span class="sr-only">next episode</span></a>
      
    </h3>
  </div>
</div>

<article>
<div class="row">
  <div class="col-md-1">
  </div>
  <div class="col-md-10">
    <h1 class="maintitle">Assembly Mapping and Binning</h1>
  </div>
  <div class="col-md-1">
  </div>
</div>


<blockquote class="objectives">
  <h2>Overview</h2>

  <div class="row">
    <div class="col-md-3">
      <strong>Teaching:</strong> 10 min
      <br/>
      <strong>Exercises:</strong> 40 min
    </div>
    <div class="col-md-9">
      <strong>Questions</strong>
      <ul>
	
	<li><p>How can I begin to analyse my metagenome?</p>
</li>
	
      </ul>
    </div>
  </div>

  <div class="row">
    <div class="col-md-3">
    </div>
    <div class="col-md-9">
      <strong>Objectives</strong>
      <ul>
	
	<li><p>Use divide and conquer approaches to break your metagenome up into your data into sections</p>
</li>
	
	<li><p>Use a metagenome binning package to bin contigs on the basis of DNA characteristics</p>
</li>
	
	<li><p>Use a metagenome binning assesment tool to evaluate the taxonomic composition of your metagenome bins</p>
</li>
	
      </ul>
    </div>
  </div>

</blockquote>

<h3 id="read-mapping-prepare-for-binning-and-coverage-assessment">Read mapping, prepare for binning and coverage assessment</h3>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>bowtie2-build final.contigs.fa contigs

bowtie2 <span class="nt">-x</span> contigs <span class="nt">-1</span> ../mySamplefiltered_1P.fq.gz <span class="nt">-2</span> ../mySamplefiltered_1P.fq.gz <span class="nt">-U</span> ../mySamplefiltered_12U.fq.gz <span class="nt">-S</span> contigs.reads.mapped.sam
</code></pre></div></div>

<p>convert sam to bam file for MetaBat</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>samtools view <span class="nt">-bS</span> contigs.reads.mapped.sam | samtools <span class="nb">sort</span> - contigs.sorted.bam
</code></pre></div></div>

<h3 id="metabat-binning">MetaBat binning</h3>

<p>There are numerous options to optimise the binning procedure. There are several tutorials
available through the metabat site. There are also numerous different metagenome binning programs avialable</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>/usr/local/anaconda2/bin/runMetaBat.sh <span class="nt">-t</span> 4 final.contigs.fa contigs.sorted.bam
</code></pre></div></div>

<div class="output highlighter-rouge"><div class="highlight"><pre class="highlight"><code>
Executing: 'jgi_summarize_bam_contig_depths --outputDepth final.contigs.fa.depth.txt --pairedContigs final.contigs.fa.paired.txt --minContigLength 1000 --minContigDepth 1  contigs.sorted.bam' at Sunday 14 October  20:45:39 AEDT 2018
Output depth matrix to final.contigs.fa.depth.txt
Output pairedContigs lower triangle to final.contigs.fa.paired.txt
minContigLength: 1000
minContigDepth: 1
Output matrix to final.contigs.fa.depth.txt
Opening 1 bams
Consolidating headers
Allocating pairedContigs matrix: 50 MB over 1 threads
Processing bam files
Thread 0 finished: contigs.sorted.bam with 21505375 reads and 4144361 readsWellMapped
Creating depth matrix file: final.contigs.fa.depth.txt
Closing most bam files
Creating pairedContigs matrix file: final.contigs.fa.paired.txt
Closing last bam file
Finished
Finished jgi_summarize_bam_contig_depths at Sunday 14 October  21:05:22 AEDT 2018
Creating depth file for metabat at Sunday 14 October  21:05:22 AEDT 2018
Executing: 'metabat2  -t 4 --inFile final.contigs.fa --outFile final.contigs.fa.metabat-bins4/bin --abdFile final.contigs.fa.depth.txt' at Sunday 14 October  21:05:22 AEDT 2018
MetaBAT 2 (v2.12.1) using minContig 2500, minCV 1.0, minCVSum 1.0, maxP 95%, minS 60, and maxEdges 200.
26 bins (40709463 bases in total) formed.
Finished metabat2 at Sunday 14 October  21:15:32 AEDT 2018
[1]+  Done                    gzip *fq  (wd: ~)
(wd now: ~/assembly5)

</code></pre></div></div>

<p>mv bins to bins directory</p>

<p>First find the output directory where your bins are stored</p>
<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>
<span class="nb">ls</span>

</code></pre></div></div>

<div class="output highlighter-rouge"><div class="highlight"><pre class="highlight"><code>contig.stats  final.contigs.fa final.contigs.fa.metabat-bins4  intermediate_contigs
contigs.sorted.bam  done  final.contigs.fa.depth.txt  final.contigs.fa.paired.txt
</code></pre></div></div>

<p>and change its name so that we are all on the same page</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nb">mv </span>final.contigs.fa.metabat-bins4 metabat.bins

</code></pre></div></div>

<p>Analyse bins using CheckM.</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>checkm lineage_wf <span class="nt">-f</span> metabat.bins/checkm.txt <span class="nt">-t</span> 4 <span class="nt">-x</span> fa metabat.bins/ metabat.bins/scg
</code></pre></div></div>

<p>When this analysis is done open and explore the results</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>nano metabat.bins/checkm.txt
</code></pre></div></div>

<p>Your output should look similar to this</p>

<p><img src="../fig/checkm.png" alt="checkm_out" /></p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>checkm bin_qa_plot <span class="nt">-x</span> fa checkm metabat.bins plots
</code></pre></div></div>

<p>Note checkm can plot a lot of metrics. If you have time, check the manual and try to produce some plots to visualise the size and quality of the binned contigs.</p>

<p>You now have a set (10-30, or more) of binned contigs with some descriptive taxonomic information that can be used to inform the next steps of your analyses.</p>

<blockquote class="challenge">
  <h2 id="question">Question:</h2>

  <p>Which bins should we keep for downstream analysis?</p>

  <p>Are there any near-complete genomes with long contigs and good coverage?</p>

  <p>Can you already being to formulate hypotheses about the source of your metagenome based on the collective taxonomic composition of the metagenome bins?</p>

</blockquote>

<p>This recent paper provides some relevant background reading <a href="https://www.nature.com/articles/sdata2017203">The reconstruction of 2,631 draft metagenome-assembled genomes from the global oceans</a></p>

<p>The workflow for this paper was published on  <a href="https://www.protocols.io/view/assembly-procedure-applied-to-tara-oceans-data-ex-hfqb3mw">protocols.io</a> so that the research could be transparently reproduced, and the code reused.</p>

<blockquote class="callout">
  <h2 id="reproducible-research">Reproducible research</h2>

  <p>Why is reproducibility important in data science?</p>

  <p>There are a number of tools that we can start to use now to help document the commands and
rationale for each step in your workflow. Some common examples include “markdown” formatted notebooks
in python (e.g. Jupyter notebook) and R notebooks available through R studio.</p>

  <p>Github is a popular tool that software developers use to develop and document programming code.
Github provides a public repository that implements  “version control”. Version control  and collprovides</p>

  <p>You are strongly encouraged to get started with Github now so that you can make use of this resource
during the next year of your studies. You may be surprised how useful it is! For example
There are several current MRes students taking advantage of the preformatted Latex template to write
their MRes Theses. This allows them to efficiently track their changes and receive feedback
on their writing from their supervisors.</p>

  <p>You might be asking “How do I get started?”. Why not start with the <a href="https://guides.github.com/activities/hello-world/">“Hello World” tutorial</a>
and then start a repository to keep a copy of your metagenomics workflow.</p>

</blockquote>

<h2 id="next-steps-detailed-annotation-of-your-metagenome-bins">Next steps: Detailed annotation of your metagenome bins</h2>

<p>We have now split our data up into 10-30, or more smaller data sets. Now is a good time to stop and think about how we are going to analyse them allows
without repeating our commands over and over and over!</p>



<blockquote class="keypoints">
  <h2>Key Points</h2>
  <ul>
    
    <li><p>A metagenome assembly combines overlapping sequences into contiguous sections of DNA called contigs</p>
</li>
    
    <li><p>An assembly is an effective way of reducing our data</p>
</li>
    
  </ul>
</blockquote>

</article>
















<div class="row">
  <div class="col-xs-1">
    <h3 class="text-left">
      
      <a href="../07-meta-genome_assembly/index.html"><span class="glyphicon glyphicon-menu-left" aria-hidden="true"></span><span class="sr-only">previous episode</span></a>
      
    </h3>
  </div>
  <div class="col-xs-10">
    
  </div>
  <div class="col-xs-1">
    <h3 class="text-right">
      
      <a href="../09-annotation-of-genome-bins/index.html"><span class="glyphicon glyphicon-menu-right" aria-hidden="true"></span><span class="sr-only">next episode</span></a>
      
    </h3>
  </div>
</div>


      
      
<footer>
  <div class="row">
    <div class="col-md-6 copyright" align="left">

    </div>
    <div class="col-md-6 help-links" align="right">
	
	<a href="/edit/gh-pages/_episodes/08-assembly_mapping_and binning.md">Edit on GitHub</a>
	
	/
	<a href="/blob/gh-pages/CONTRIBUTING.md">Contributing</a>
	/
	<a href="/">Source</a>
	/
	<a href="/blob/gh-pages/CITATION">Cite</a>
	/
	<a href="mailto:martin.ostrowski@mq.edu.au">Contact</a>
    </div>
  </div>
  <div class="row">
    <div class="col-md-12" align="center">
      Using <a href="https://github.com/carpentries/styles/">The Carpentries style</a>
      version <a href="https://github.com/carpentries/styles/releases/tag/v9.5.3">9.5.3</a>.
    </div>
  </div>
</footer>

      
    </div>
    
<script src="../assets/js/jquery.min.js"></script>
<script src="../assets/js/bootstrap.min.js"></script>
<script src="../assets/js/lesson.js"></script>
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
  ga('create', 'UA-37305346-2', 'auto');
  ga('send', 'pageview');
</script>

  </body>
</html>
