















<!doctype html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta http-equiv="last-modified" content="2018-10-20 22:53:04 +1100">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <!-- meta "search-domain" used for google site search function google_search() -->
    <meta name="search-domain" value="">
    <link rel="stylesheet" type="text/css" href="../assets/css/bootstrap.css" />
    <link rel="stylesheet" type="text/css" href="../assets/css/bootstrap-theme.css" />
    <link rel="stylesheet" type="text/css" href="../assets/css/lesson.css" />
    <link rel="stylesheet" type="text/css" href="../assets/css/syntax.css" />


    <!-- HTML5 shim and Respond.js for IE8 support of HTML5 elements and media queries -->
    <!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
    <!--[if lt IE 9]>
	<script src="https://oss.maxcdn.com/html5shiv/3.7.2/html5shiv.min.js"></script>
	<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
	<![endif]-->
    <title>CBMS Advanced Biomolecular Analysis - Genomics: Genome Assembly Workflow</title>
  </head>
  <body>

    







    <div class="container">
      
<nav class="navbar navbar-default">
  <div class="container-fluid">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>



      
      <a class="navbar-brand" href="../index.html">Home</a>

    </div>
    <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
      <ul class="nav navbar-nav">


        
	
        <li><a href="../setup.html">Setup</a></li>

        
        <li class="dropdown">
          <a href="../" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">Episodes <span class="caret"></span></a>
          <ul class="dropdown-menu">
            
            <li><a href="../01-intro/index.html">Introducing the Shell</a></li>
            
            <li><a href="../02-filedir/index.html">Navigating Files and Directories</a></li>
            
            <li><a href="../03-create/index.html">Working With Files and Directories</a></li>
            
            <li><a href="../04-pipefilter/index.html">Pipes and Filters</a></li>
            
            <li><a href="../06-Genome_assembly/index.html">Genome Assembly Workflow</a></li>
            
            <li><a href="../06-trimming/index.html">Trimming and Filtering</a></li>
            
            <li><a href="../07-meta-genome_assembly/index.html">Genome Assembly Workflow - Week 2</a></li>
            
            <li><a href="../08-assembly_mapping_and%20binning/index.html">Assembly Mapping and Binning</a></li>
            
            <li><a href="../09-annotation-of-genome-bins/index.html">Annotation of genome bins. Week 2-3</a></li>
            
            <li><a href="../12-taxonomic-assignment/index.html">Taxonomic assignment using LCA. Week 3-4</a></li>
            
	    <li role="separator" class="divider"></li>
            <li><a href="../aio.html">All in one page (Beta)</a></li>
          </ul>
        </li>
	

	
	
        <li class="dropdown">
          <a href="../" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">Extras <span class="caret"></span></a>
          <ul class="dropdown-menu">
            <li><a href="../reference.html">Reference</a></li>
            
            <li><a href="../about/index.html">About</a></li>
            
            <li><a href="../discuss/index.html">Discussion</a></li>
            
            <li><a href="../figures/index.html">Figures</a></li>
            
            <li><a href="../guide/index.html">Instructor Notes</a></li>
            
          </ul>
        </li>
	

	
        <li><a href="../LICENSE.html">License</a></li>
	
	
	<li><a href="/edit/gh-pages/_episodes_rmd/06-Genome_assembly.Rmd">Improve this page <span class="glyphicon glyphicon-pencil" aria-hidden="true"></span></a></li>
	
	
      </ul>
      <form class="navbar-form navbar-right" role="search" id="search" onsubmit="google_search(); return false;">
        <div class="form-group">
          <input type="text" id="google-search" placeholder="Search..." aria-label="Google site search">
        </div>
      </form>
    </div>
  </div>
</nav>

















<div class="row">
  <div class="col-xs-1">
    <h3 class="text-left">
      
      <a href="../04-pipefilter/index.html"><span class="glyphicon glyphicon-menu-left" aria-hidden="true"></span><span class="sr-only">previous episode</span></a>
      
    </h3>
  </div>
  <div class="col-xs-10">
    
    <h3 class="maintitle"><a href="../">CBMS Advanced Biomolecular Analysis - Genomics</a></h3>
    
  </div>
  <div class="col-xs-1">
    <h3 class="text-right">
      
      <a href="../06-trimming/index.html"><span class="glyphicon glyphicon-menu-right" aria-hidden="true"></span><span class="sr-only">next episode</span></a>
      
    </h3>
  </div>
</div>

<article>
<div class="row">
  <div class="col-md-1">
  </div>
  <div class="col-md-10">
    <h1 class="maintitle">Genome Assembly Workflow</h1>
  </div>
  <div class="col-md-1">
  </div>
</div>


<blockquote class="objectives">
  <h2>Overview</h2>

  <div class="row">
    <div class="col-md-3">
      <strong>Teaching:</strong> 10 min
      <br/>
      <strong>Exercises:</strong> 40 min
    </div>
    <div class="col-md-9">
      <strong>Questions</strong>
      <ul>
	
	<li><p>How to quality control raw sequence files for genome assembly</p>
</li>
	
      </ul>
    </div>
  </div>

  <div class="row">
    <div class="col-md-3">
    </div>
    <div class="col-md-9">
      <strong>Objectives</strong>
      <ul>
	
	<li><p>Describe in detail the quality measures of your raw sequences</p>
</li>
	
	<li><p>Explain the key points in the genome assembly workflow</p>
</li>
	
	<li><p>Demonstrate how to include pieces of code, figures, and challenges</p>
</li>
	
      </ul>
    </div>
  </div>

</blockquote>

<h2 id="metagenome-assembly-illumina-paired-end-sequences">Metagenome assembly, illumina Paired end sequences</h2>

<p>In your directory you will find two, or more, sequence files <code class="highlighter-rouge">.fq.gz</code> in <em>fastq</em> format. We are going
to discuss what is in each of these files, begin to understand how they are formatted and explore different ways to quality check them.</p>

<p>The following notes are not intended to be a complete workflow. Where necessary you will have to fill the gaps <code class="highlighter-rouge">xxx</code> using
relevant filenames and tools you have already learned, or by doing some research on the web.</p>

<blockquote class="challenge">
  <h2 id="questions">Questions:</h2>

  <p>What basic bash commands can you use to determine the size, and number of sequences in your fastq files?</p>

  <blockquote class="language-bash solution">
    <h2 id="solution">Solution</h2>
    <p>ls -la</p>

    <div class="output highlighter-rouge"><div class="highlight"><pre class="highlight"><code>ls -la

 gunzip *

 less xxx

fastqc xxx
</code></pre></div>    </div>
  </blockquote>
</blockquote>

<p>Explore the fastqc output</p>

<blockquote class="challenge">
  <h2 id="questions-1">Questions:</h2>

  <p>How many sequences are in your files?</p>

  <p>How does the quality profile of the R1 and R2 reads compare?</p>

  <p>Do your sequences need to be trimmed because the quality issues or leftover adaptor sequences?</p>

  <p>What options do you have for viewing the output of the fastqc analyses?</p>

</blockquote>

<p>If your sequences requires filtering we are going to use the program trimmomatic using the following code</p>
<ol>
  <li>trim R1 and R2 using trimmomatic, edit the input and output names to something rational</li>
</ol>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>java <span class="nt">-jar</span> /usr/local/Trimmomatic-0.38/trimmomatic-0.38.jar PE <span class="nt">-threads</span> 4 YON20160503d0_R1_001.fq.gz YON20160503d0_R2_001.fq.gz <span class="nt">-baseout</span> mySamplefiltered.fq.gz ILLUMINACLIP:/usr/local/Trimmomatic-0.38/adapters/NexteraPE-PE.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36
</code></pre></div></div>

<blockquote class="challenge">
  <h2 id="questions-2">Questions:</h2>

  <p>How does the quantity and quality of the sequences in your filtered files compare to your raw sequence files?</p>

</blockquote>

<p>Next, and final, task for the day is to start the metagenome assembly using a popular assembly program (spades or megahit)</p>

<p>It is likely that this will take sometime so we will need to setup a special terminal session that will not be interrupted when you log out.
We will use the program screen and give this session a name</p>

<ol>
  <li>Join the two unpaired reads files and start the spades assembly (edit the names)</li>
</ol>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>screen <span class="nt">-S</span> Assembly

<span class="nb">cat </span>mySamplefiltered_<span class="k">*</span>U.fq.gz <span class="o">&gt;</span> mySamplefiltered_12U.fq.gz

/usr/local/megahit/megahit <span class="nt">-1</span> mySamplefiltered_1P.fq.gz <span class="nt">-2</span> mySamplefiltered_2P.fq.gz <span class="nt">-r</span> mySamplefiltered_12U.fq.gz <span class="nt">-t</span> 4 <span class="nt">-o</span> assembly
</code></pre></div></div>

<p>##That is all for this week</p>

<p>Produce a summary of results using quast</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>/usr/local/quast/quast-4.6.3/quast.py contigs.fa
</code></pre></div></div>

<blockquote class="challenge">
  <h2 id="questions-3">Questions:</h2>

  <p>How does the total size of your input data compare to size of your assembly?</p>

  <p>How long is the longest contig?</p>

  <p>Which contig is assembled from the greatest number of input sequences?</p>

  <blockquote class="language-bash solution">
    <h2 id="solutions">Solutions</h2>

    <div class="output highlighter-rouge"><div class="highlight"><pre class="highlight"><code>
</code></pre></div>    </div>
  </blockquote>
</blockquote>

<ol>
  <li>read mapping, prepare for binning and coverage assessment</li>
</ol>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>bowtie2-build final.contigs.fa scaffolds

bowtie2 <span class="nt">-x</span> contigs <span class="nt">-1</span> ../mySamplefiltered_1P.fq.gz <span class="nt">-2</span> ../AmySamplefiltered_1P.fq.gz <span class="nt">-U</span> ../mySamplefiltered_12U.fq.gz <span class="nt">-S</span> contigs.reads.mapped.sam
</code></pre></div></div>

<p>convert sam to bam file for MetaBat # this needs to be fixed</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>samtools view <span class="nt">-bS</span> contigs.reads.mapped.sam | samtools <span class="nb">sort</span> - contigs.sorted
</code></pre></div></div>

<ol>
  <li>MetaBat binning</li>
</ol>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>/usr/local/anaconda2/bin/runMetaBat.sh <span class="nt">-t</span> 4 final.contigs.fa contigs.sorted.bam
</code></pre></div></div>

<p>mv bins to bins directory !!</p>

<p>and change its name so that we are all on the same page</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nb">mv </span>xxxxxxxxx metabat.bins

</code></pre></div></div>

<p>Analyse bins using CheckM.</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>checkm lineage_wf <span class="nt">-f</span> metabat.bins/checkm.txt <span class="nt">-t</span> 4 <span class="nt">-x</span> fa metabat.bins/ metabat.bins/scg
</code></pre></div></div>

<p>When this analysis is done open and explore the results</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>nano metabat.bins/checkm.txt
</code></pre></div></div>



<blockquote class="keypoints">
  <h2>Key Points</h2>
  <ul>
    
    <li><p>Garbage in, Garbage out! Focus on quality, and methods to improve the input data</p>
</li>
    
  </ul>
</blockquote>

</article>
















<div class="row">
  <div class="col-xs-1">
    <h3 class="text-left">
      
      <a href="../04-pipefilter/index.html"><span class="glyphicon glyphicon-menu-left" aria-hidden="true"></span><span class="sr-only">previous episode</span></a>
      
    </h3>
  </div>
  <div class="col-xs-10">
    
  </div>
  <div class="col-xs-1">
    <h3 class="text-right">
      
      <a href="../06-trimming/index.html"><span class="glyphicon glyphicon-menu-right" aria-hidden="true"></span><span class="sr-only">next episode</span></a>
      
    </h3>
  </div>
</div>


      
      
<footer>
  <div class="row">
    <div class="col-md-6 copyright" align="left">

    </div>
    <div class="col-md-6 help-links" align="right">
	
	
	<a href="/edit/gh-pages/_episodes_rmd/06-Genome_assembly.Rmd">Edit on GitHub</a>
	
	
	/
	<a href="/blob/gh-pages/CONTRIBUTING.md">Contributing</a>
	/
	<a href="/">Source</a>
	/
	<a href="/blob/gh-pages/CITATION">Cite</a>
	/
	<a href="mailto:martin.ostrowski@mq.edu.au">Contact</a>
    </div>
  </div>
  <div class="row">
    <div class="col-md-12" align="center">
      Using <a href="https://github.com/carpentries/styles/">The Carpentries style</a>
      version <a href="https://github.com/carpentries/styles/releases/tag/v9.5.3">9.5.3</a>.
    </div>
  </div>
</footer>

      
    </div>
    
<script src="../assets/js/jquery.min.js"></script>
<script src="../assets/js/bootstrap.min.js"></script>
<script src="../assets/js/lesson.js"></script>
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
  ga('create', 'UA-37305346-2', 'auto');
  ga('send', 'pageview');
</script>

  </body>
</html>
