















<!doctype html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta http-equiv="last-modified" content="2018-10-20 22:53:04 +1100">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <!-- meta "search-domain" used for google site search function google_search() -->
    <meta name="search-domain" value="">
    <link rel="stylesheet" type="text/css" href="../assets/css/bootstrap.css" />
    <link rel="stylesheet" type="text/css" href="../assets/css/bootstrap-theme.css" />
    <link rel="stylesheet" type="text/css" href="../assets/css/lesson.css" />
    <link rel="stylesheet" type="text/css" href="../assets/css/syntax.css" />


    <!-- HTML5 shim and Respond.js for IE8 support of HTML5 elements and media queries -->
    <!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
    <!--[if lt IE 9]>
	<script src="https://oss.maxcdn.com/html5shiv/3.7.2/html5shiv.min.js"></script>
	<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
	<![endif]-->
    <title>CBMS Advanced Biomolecular Analysis - Genomics: Genome Assembly Workflow - Week 2</title>
  </head>
  <body>

    







    <div class="container">
      
<nav class="navbar navbar-default">
  <div class="container-fluid">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>



      
      <a class="navbar-brand" href="../index.html">Home</a>

    </div>
    <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
      <ul class="nav navbar-nav">


        
	
        <li><a href="../setup.html">Setup</a></li>

        
        <li class="dropdown">
          <a href="../" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">Episodes <span class="caret"></span></a>
          <ul class="dropdown-menu">
            
            <li><a href="../01-intro/index.html">Introducing the Shell</a></li>
            
            <li><a href="../02-filedir/index.html">Navigating Files and Directories</a></li>
            
            <li><a href="../03-create/index.html">Working With Files and Directories</a></li>
            
            <li><a href="../04-pipefilter/index.html">Pipes and Filters</a></li>
            
            <li><a href="../06-Genome_assembly/index.html">Genome Assembly Workflow</a></li>
            
            <li><a href="../06-trimming/index.html">Trimming and Filtering</a></li>
            
            <li><a href="../07-meta-genome_assembly/index.html">Genome Assembly Workflow - Week 2</a></li>
            
            <li><a href="../08-assembly_mapping_and%20binning/index.html">Assembly Mapping and Binning</a></li>
            
            <li><a href="../09-annotation-of-genome-bins/index.html">Annotation of genome bins. Week 2-3</a></li>
            
            <li><a href="../12-taxonomic-assignment/index.html">Taxonomic assignment using LCA. Week 3-4</a></li>
            
	    <li role="separator" class="divider"></li>
            <li><a href="../aio.html">All in one page (Beta)</a></li>
          </ul>
        </li>
	

	
	
        <li class="dropdown">
          <a href="../" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">Extras <span class="caret"></span></a>
          <ul class="dropdown-menu">
            <li><a href="../reference.html">Reference</a></li>
            
            <li><a href="../about/index.html">About</a></li>
            
            <li><a href="../discuss/index.html">Discussion</a></li>
            
            <li><a href="../figures/index.html">Figures</a></li>
            
            <li><a href="../guide/index.html">Instructor Notes</a></li>
            
          </ul>
        </li>
	

	
        <li><a href="../LICENSE.html">License</a></li>
	
	<li><a href="/edit/gh-pages/_episodes/07-meta-genome_assembly.md">Improve this page <span class="glyphicon glyphicon-pencil" aria-hidden="true"></span></a></li>
	
      </ul>
      <form class="navbar-form navbar-right" role="search" id="search" onsubmit="google_search(); return false;">
        <div class="form-group">
          <input type="text" id="google-search" placeholder="Search..." aria-label="Google site search">
        </div>
      </form>
    </div>
  </div>
</nav>

















<div class="row">
  <div class="col-xs-1">
    <h3 class="text-left">
      
      <a href="../06-trimming/index.html"><span class="glyphicon glyphicon-menu-left" aria-hidden="true"></span><span class="sr-only">previous episode</span></a>
      
    </h3>
  </div>
  <div class="col-xs-10">
    
    <h3 class="maintitle"><a href="../">CBMS Advanced Biomolecular Analysis - Genomics</a></h3>
    
  </div>
  <div class="col-xs-1">
    <h3 class="text-right">
      
      <a href="../08-assembly_mapping_and%20binning/index.html"><span class="glyphicon glyphicon-menu-right" aria-hidden="true"></span><span class="sr-only">next episode</span></a>
      
    </h3>
  </div>
</div>

<article>
<div class="row">
  <div class="col-md-1">
  </div>
  <div class="col-md-10">
    <h1 class="maintitle">Genome Assembly Workflow - Week 2</h1>
  </div>
  <div class="col-md-1">
  </div>
</div>


<blockquote class="objectives">
  <h2>Overview</h2>

  <div class="row">
    <div class="col-md-3">
      <strong>Teaching:</strong> 10 min
      <br/>
      <strong>Exercises:</strong> 40 min
    </div>
    <div class="col-md-9">
      <strong>Questions</strong>
      <ul>
	
	<li><p>How to develop a metagenomics analysis workflow?</p>
</li>
	
      </ul>
    </div>
  </div>

  <div class="row">
    <div class="col-md-3">
    </div>
    <div class="col-md-9">
      <strong>Objectives</strong>
      <ul>
	
	<li><p>Develop an outline and examine the key points in a genome assembly and annotation workflow</p>
</li>
	
	<li><p>Describe in detail the quality measures of your assembled contigs</p>
</li>
	
	<li><p>Begin to analyse you assembled metagenome using binning, mapping and database comparisons</p>
</li>
	
	<li><p>Use shell tools to evaluate the results of your analyses</p>
</li>
	
      </ul>
    </div>
  </div>

</blockquote>

<h2 id="completed-metagenome-assembly-illumina-paired-end-sequences">Completed Metagenome assembly, illumina Paired end sequences</h2>

<p>During the first class we trimmed and filtered a pair of sequence files <code class="highlighter-rouge">.fq.gz</code> in <em>fastq</em> format.
We used tools available in the BASH terminal to rapidly reveal some details about the contents of these
files and we used the program <code class="highlighter-rouge">fastqc</code> to provide a detailed overview of the quality and composition of the sequences
within them. <code class="highlighter-rouge">Trimmomatic</code> was used to remove any leftover adaptor sequences and to trim the end of sequences
that fell below a quality threshold.</p>

<h3 id="bioinformatics-workflows">Bioinformatics workflows</h3>

<p>When working with high-throughput sequencing data, the raw reads you get off of the sequencer will need to pass
through a number of  different tools in order to generate your final desired output. The execution of this set of
tools in a specified order is commonly referred to as a <em>workflow</em> or a <em>pipeline</em>.</p>

<p>An example of the workflow we will be using for our metagenome assembly, binning and annotation exercises is given below</p>

<p><img src="../fig/workflow.png" alt="workflow" /></p>

<ol>
  <li>Quality control - Assessing quality using FastQC</li>
  <li>Quality control - Trimming and/or filtering reads (if necessary)</li>
  <li>Assemble a Metagenome/Genome or Align reads to reference genome</li>
  <li>Perform post-alignment processes such as binning, gene calling and annotation</li>
</ol>

<h3 id="week-1-review-quality-assessment-and-trimming">Week 1 review: Quality Assessment and trimming</h3>

<p>The first step in the Metagenome assembly and annotation workflow was to take the FASTQ files received from the sequencing facility
and assess the quality of the sequence reads.</p>

<h3 id="a-refresher-on-the-fastq-format">A refresher on the FASTQ format</h3>

<p>Although it looks complicated (and it is), it’s easy to understand the
<a href="https://en.wikipedia.org/wiki/FASTQ_format">fastq</a> format with a little decoding. Some rules about the format
include…</p>

<table>
  <thead>
    <tr>
      <th>Line</th>
      <th>Description</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>1</td>
      <td>Always begins with ‘@’ and then information about the read</td>
    </tr>
    <tr>
      <td>2</td>
      <td>The actual DNA sequence</td>
    </tr>
    <tr>
      <td>3</td>
      <td>Always begins with a ‘+’ and sometimes the same info in line 1</td>
    </tr>
    <tr>
      <td>4</td>
      <td>Has a string of characters which represent the quality scores; must have same number of characters as line 2</td>
    </tr>
  </tbody>
</table>

<p>We can view the first complete read in one of the files our dataset by using <code class="highlighter-rouge">head</code> to look at
the first four lines in any of your fastq (or fq) files.</p>

<div class="bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>$ head -n 4 *L001_R1.fastq
</code></pre></div></div>

<div class="output highlighter-rouge"><div class="highlight"><pre class="highlight"><code>
@D00626:337:H5VJHBCX2:1:1106:1162:2154 1:N:0:AAGAGGCA+GTAAGGAG
GACCTAAGATACAAATCTTAGAATCAAGATCATGCTCATCTTTTAGCTTTGTTTGGATATTATAACTCATTCCGCTCCTGTGCAAACACCAGTTGTCAGAGCAATCGCTGATGATCTTCTTCCCATGAC
+
DDDDDIIHIIIIGIIIIIIIIIGHIIIIIIIIHIIGHIHIHHHIHGHIHIIHIHHGIHIIIIIIIIIGIIIIIIIIIIIIEFHHHHIIIIFIIEHHHFEHHFHHIEHIHIIIHIIIIHIHHIGFHI?HG
</code></pre></div></div>

<p>Now compare a reverse read in the R2 file.</p>
<div class="bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>head -n 4 *_L001_R2.fastq
</code></pre></div></div>

<div class="output highlighter-rouge"><div class="highlight"><pre class="highlight"><code>@D00626:337:H5VJHBCX2:1:1106:1162:2154 2:N:0:AAGAGGCA+GTAAGGAG
GCTTCAAGTCATGATCGAAACTTTATTATTGAAGTCATGGGAAGAAGATCATCAGCGATTGCTCTGANNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNTNNNNNNNCNNNNNNNNANGATNNNN
+
@BADDIIIIIIIIIIIIIIIGHIIIIIIIHIII?GHIHIIIHIHIHHHHHIHGHHIIIIIIHIIIII#####################################-#######-########8#-8-####

</code></pre></div></div>

<p>Do you notice the run of Ns at the end where the quality of the sequence was not good.</p>

<h3 id="picking-up-where-we-left-off">Picking up where we left off</h3>

<p>We concluded the first class by establishing a special terminal session with <code class="highlighter-rouge">screen</code> so that the
metagenome assembly program could run without interruption overnight, or over the weekend. <a href="https://bitbucket.org/berkeleylab/metabat/overview">Megahit</a> was our
chosen assembly program.</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>screen <span class="nt">-S</span> Assembly

<span class="nb">cat </span>mySamplefiltered_<span class="k">*</span>U.fq.gz <span class="o">&gt;</span> mySamplefiltered_12U.fq.gz

/usr/local/megahit/megahit <span class="nt">-1</span> mySamplefiltered_1P.fq.gz <span class="nt">-2</span> mySamplefiltered_2P.fq.gz <span class="nt">-r</span> mySamplefiltered_12U.fq.gz <span class="nt">-t</span> 4 <span class="nt">-o</span> assembly
</code></pre></div></div>

<p>By now you should have a completed metgenome assembly in the assembly directory (or possibly not?).</p>

<p>Login to your account on the server and recover your <code class="highlighter-rouge">screen</code><em>ed</em> terminal session.</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>screen <span class="nt">-r</span>
</code></pre></div></div>

<p>If you have multiple screen sessions detached you may be presented with a list</p>

<div class="output highlighter-rouge"><div class="highlight"><pre class="highlight"><code>
There are several suitable screens on:
	88323.assembly	(14/10/18 11:25:56)	(Detached)
	16277.superfocus	(14/10/18 09:53:02)	(Detached)
Type "screen [-d] -r [pid.]tty.host" to resume one of them.
</code></pre></div></div>

<p>If this is the case select which screened session you would like to resume</p>
<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>screen <span class="nt">-r</span> 88323.assembly
</code></pre></div></div>

<p>Let us examine the results</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nb">cd </span>assembly

<span class="nb">ls</span>
</code></pre></div></div>

<p>If the assembly completed without error the contents of your assembly directory look like
this and it is time to start exploring the output</p>

<div class="output highlighter-rouge"><div class="highlight"><pre class="highlight"><code>done final.contigs.fa Intermediate_contigs log
</code></pre></div></div>

<blockquote class="challenge">
  <h2 id="questions">Questions:</h2>

  <p>What did this assembly program actually do?</p>

  <p>How many sequences are in your files?</p>

  <p>What information can we extract from the final.contigs.fa file?</p>

  <p>How does the quantity and quality of the sequences in your assembled contigs compare to your raw sequence files?</p>

  <blockquote class="solution">
    <h2 id="suggestions">Suggestions</h2>

    <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nb">grep</span> <span class="nt">-c</span> <span class="s1">'&gt;'</span> final.contigs.fa
</code></pre></div>    </div>
    <div class="output highlighter-rouge"><div class="highlight"><pre class="highlight"><code>xxxx
</code></pre></div>    </div>
    <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nb">grep</span> <span class="s1">'&gt;'</span> final.contigs.fa | less
</code></pre></div>    </div>
    <div class="output highlighter-rouge"><div class="highlight"><pre class="highlight"><code>xxxx
</code></pre></div>    </div>
    <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nb">cat </span>final.contigs.fa  | <span class="nb">grep</span> <span class="s1">'&gt;'</span> | <span class="nb">cut</span> <span class="nt">-f</span> 4 <span class="nt">-d</span><span class="s1">' '</span> | <span class="nb">sort</span> <span class="nt">-n</span>
</code></pre></div>    </div>
    <div class="output highlighter-rouge"><div class="highlight"><pre class="highlight"><code>xxxx
</code></pre></div>    </div>
    <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nb">tail </span>log
</code></pre></div>    </div>
    <div class="output highlighter-rouge"><div class="highlight"><pre class="highlight"><code>xxxx
</code></pre></div>    </div>
  </blockquote>
</blockquote>

<blockquote class="callout">
  <h2 id="what-is-a-contig">What is a contig?</h2>

  <p>A contig (from contiguous) is a set of overlapping DNA segments that together represent a consensus region of DNA.</p>

</blockquote>

<h2 id="assessment-of-your-assembly-in-more-detail">Assessment of your assembly in more detail</h2>

<p>Next we will use quast, a program to produce a detailed summary of our metagenome assembly</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>/usr/local/quast/quast-4.6.3/quast.py final.contigs.fa
</code></pre></div></div>

<p>Quast will produce a two reports, one in plain text and one in .html format. View which ever report you prefer and
start compiling a table of key parameters describing your metagenome assembly. Use the questions below as a guide.</p>

<blockquote class="challenge">
  <h2 id="questions-1">Questions:</h2>

  <p>How does the total size of your input data compare to size of your assembly?</p>

  <p>How long is the longest contig?</p>

  <p>Which contig is assembled from the greatest number of input sequences?</p>

  <p>What is the N50? How does your N50 compare with your neighbor?</p>

  <blockquote class="language-bash solution">
    <h2 id="solutions">Solutions</h2>

    <div class="output highlighter-rouge"><div class="highlight"><pre class="highlight"><code>
</code></pre></div>    </div>
  </blockquote>
</blockquote>

<p>Make a table to track what has happened to the quantity, quality and total length of your sequences as they progress from raw to filtered to assembled.</p>



<blockquote class="keypoints">
  <h2>Key Points</h2>
  <ul>
    
    <li><p>Develop a plan. Evaluate the quality of your processing outputs at every stage</p>
</li>
    
    <li><p>A metagenome assembly combines overlapping sequences into contiguous sections of DNA called contigs</p>
</li>
    
    <li><p>An assembly is an effective way or reducing our data</p>
</li>
    
  </ul>
</blockquote>

</article>
















<div class="row">
  <div class="col-xs-1">
    <h3 class="text-left">
      
      <a href="../06-trimming/index.html"><span class="glyphicon glyphicon-menu-left" aria-hidden="true"></span><span class="sr-only">previous episode</span></a>
      
    </h3>
  </div>
  <div class="col-xs-10">
    
  </div>
  <div class="col-xs-1">
    <h3 class="text-right">
      
      <a href="../08-assembly_mapping_and%20binning/index.html"><span class="glyphicon glyphicon-menu-right" aria-hidden="true"></span><span class="sr-only">next episode</span></a>
      
    </h3>
  </div>
</div>


      
      
<footer>
  <div class="row">
    <div class="col-md-6 copyright" align="left">

    </div>
    <div class="col-md-6 help-links" align="right">
	
	<a href="/edit/gh-pages/_episodes/07-meta-genome_assembly.md">Edit on GitHub</a>
	
	/
	<a href="/blob/gh-pages/CONTRIBUTING.md">Contributing</a>
	/
	<a href="/">Source</a>
	/
	<a href="/blob/gh-pages/CITATION">Cite</a>
	/
	<a href="mailto:martin.ostrowski@mq.edu.au">Contact</a>
    </div>
  </div>
  <div class="row">
    <div class="col-md-12" align="center">
      Using <a href="https://github.com/carpentries/styles/">The Carpentries style</a>
      version <a href="https://github.com/carpentries/styles/releases/tag/v9.5.3">9.5.3</a>.
    </div>
  </div>
</footer>

      
    </div>
    
<script src="../assets/js/jquery.min.js"></script>
<script src="../assets/js/bootstrap.min.js"></script>
<script src="../assets/js/lesson.js"></script>
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
  ga('create', 'UA-37305346-2', 'auto');
  ga('send', 'pageview');
</script>

  </body>
</html>
