















<!doctype html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta http-equiv="last-modified" content="2018-10-20 22:53:04 +1100">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <!-- meta "search-domain" used for google site search function google_search() -->
    <meta name="search-domain" value="">
    <link rel="stylesheet" type="text/css" href="../assets/css/bootstrap.css" />
    <link rel="stylesheet" type="text/css" href="../assets/css/bootstrap-theme.css" />
    <link rel="stylesheet" type="text/css" href="../assets/css/lesson.css" />
    <link rel="stylesheet" type="text/css" href="../assets/css/syntax.css" />


    <!-- HTML5 shim and Respond.js for IE8 support of HTML5 elements and media queries -->
    <!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
    <!--[if lt IE 9]>
	<script src="https://oss.maxcdn.com/html5shiv/3.7.2/html5shiv.min.js"></script>
	<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
	<![endif]-->
    <title>CBMS Advanced Biomolecular Analysis - Genomics: Taxonomic assignment using LCA. Week 3-4</title>
  </head>
  <body>

    







    <div class="container">
      
<nav class="navbar navbar-default">
  <div class="container-fluid">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>



      
      <a class="navbar-brand" href="../index.html">Home</a>

    </div>
    <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
      <ul class="nav navbar-nav">


        
	
        <li><a href="../setup.html">Setup</a></li>

        
        <li class="dropdown">
          <a href="../" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">Episodes <span class="caret"></span></a>
          <ul class="dropdown-menu">
            
            <li><a href="../01-intro/index.html">Introducing the Shell</a></li>
            
            <li><a href="../02-filedir/index.html">Navigating Files and Directories</a></li>
            
            <li><a href="../03-create/index.html">Working With Files and Directories</a></li>
            
            <li><a href="../04-pipefilter/index.html">Pipes and Filters</a></li>
            
            <li><a href="../06-Genome_assembly/index.html">Genome Assembly Workflow</a></li>
            
            <li><a href="../06-trimming/index.html">Trimming and Filtering</a></li>
            
            <li><a href="../07-meta-genome_assembly/index.html">Genome Assembly Workflow - Week 2</a></li>
            
            <li><a href="../08-assembly_mapping_and%20binning/index.html">Assembly Mapping and Binning</a></li>
            
            <li><a href="../09-annotation-of-genome-bins/index.html">Annotation of genome bins. Week 2-3</a></li>
            
            <li><a href="../12-taxonomic-assignment/index.html">Taxonomic assignment using LCA. Week 3-4</a></li>
            
	    <li role="separator" class="divider"></li>
            <li><a href="../aio.html">All in one page (Beta)</a></li>
          </ul>
        </li>
	

	
	
        <li class="dropdown">
          <a href="../" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">Extras <span class="caret"></span></a>
          <ul class="dropdown-menu">
            <li><a href="../reference.html">Reference</a></li>
            
            <li><a href="../about/index.html">About</a></li>
            
            <li><a href="../discuss/index.html">Discussion</a></li>
            
            <li><a href="../figures/index.html">Figures</a></li>
            
            <li><a href="../guide/index.html">Instructor Notes</a></li>
            
          </ul>
        </li>
	

	
        <li><a href="../LICENSE.html">License</a></li>
	
	
	<li><a href="/edit/gh-pages/_episodes_rmd/12-taxonomic-assignment.Rmd">Improve this page <span class="glyphicon glyphicon-pencil" aria-hidden="true"></span></a></li>
	
	
      </ul>
      <form class="navbar-form navbar-right" role="search" id="search" onsubmit="google_search(); return false;">
        <div class="form-group">
          <input type="text" id="google-search" placeholder="Search..." aria-label="Google site search">
        </div>
      </form>
    </div>
  </div>
</nav>

















<div class="row">
  <div class="col-xs-1">
    <h3 class="text-left">
      
      <a href="../09-annotation-of-genome-bins/index.html"><span class="glyphicon glyphicon-menu-left" aria-hidden="true"></span><span class="sr-only">previous episode</span></a>
      
    </h3>
  </div>
  <div class="col-xs-10">
    
    <h3 class="maintitle"><a href="../">CBMS Advanced Biomolecular Analysis - Genomics</a></h3>
    
  </div>
  <div class="col-xs-1">
    <h3 class="text-right">
      
      <a href="../"><span class="glyphicon glyphicon-menu-up" aria-hidden="true"></span><span class="sr-only">lesson home</span></a>
      
    </h3>
  </div>
</div>

<article>
<div class="row">
  <div class="col-md-1">
  </div>
  <div class="col-md-10">
    <h1 class="maintitle">Taxonomic assignment using LCA. Week 3-4</h1>
  </div>
  <div class="col-md-1">
  </div>
</div>


<blockquote class="objectives">
  <h2>Overview</h2>

  <div class="row">
    <div class="col-md-3">
      <strong>Teaching:</strong> 5 min
      <br/>
      <strong>Exercises:</strong> 40 min
    </div>
    <div class="col-md-9">
      <strong>Questions</strong>
      <ul>
	
	<li><p>How can I assign my sequences to the nearest taxonomic lineage?</p>
</li>
	
      </ul>
    </div>
  </div>

  <div class="row">
    <div class="col-md-3">
    </div>
    <div class="col-md-9">
      <strong>Objectives</strong>
      <ul>
	
	<li><p>Identify databases and tools that could serve as sources of taxonomic information for your sequences</p>
</li>
	
	<li><p>Learn how to repeat commands and processes tens, hundreds or thousands of files using loops</p>
</li>
	
	<li><p>Extract functional information from sequence database searches</p>
</li>
	
	<li><p>Practice writing loops</p>
</li>
	
      </ul>
    </div>
  </div>

</blockquote>

<h2 id="basic-sequence-taxonomy-annotation">BAsic Sequence Taxonomy Annotation</h2>

<p><a href="https://github.com/timkahlke/BASTA/wiki">BASTA</a> is a command-line tool written in python that was developed to enable a basic taxonomy annotation based on a Last Common Ancestor algorithm. If you like BASTA and use it for publications please cite it as “Kahlke T and Ralph PJ (2018), BASTA–Taxonomic classification of sequences and sequence bins using Last Common Ancestor estimations. Meth. Ecol. Evol. doi:10.1111/2041‐210X.13095”</p>

<p>We can use BASTA to determine the most likely source of each bin, contig and sequence. We will use BASTA to evaluate the specificity of the MetaBat binning by comparing the taxonomic composition of the overall assembly, the consensus LCA estimate for each entire bin and the composition protein sequences in each bin.</p>

<p>Our input analyses will be a diamond (blastp) sequence search of the predicted protein sequences from PROKKA.</p>

<p>cd to your metabat.bins directory and rename your protein sequence predictions from prokka (.faa) while copying to a new directory</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nb">mkdir </span>prokka.faa

<span class="k">for </span>subdir <span class="k">in</span> <span class="k">*</span>prokka.out<span class="p">;</span> <span class="k">do </span><span class="nb">cp</span> <span class="nv">$subdir</span>/<span class="k">*</span>faa prokka.faa/<span class="nv">$subdir</span>.faa<span class="p">;</span> <span class="k">done</span><span class="p">;</span>
</code></pre></div></div>

<p>cd to the new directory and check the contents are what you expect</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nb">cd </span>prokka.faa

<span class="nb">ls</span>
</code></pre></div></div>

<p>There are probably a large number of bins to process so we should think about constructing a loop. Putting things together piece by piece.</p>

<p>In the example we’ll start with the predicted protein sequences in bin.10 and do a sequence search (blastp) against the NCBI non-redundant database using <a href="https://github.com/bbuchfink/diamond">diamond</a>. DIAMOND is a sequence aligner for protein and translated DNA searches, designed for high performance analysis of big sequence data.</p>

<p>The help file is always a good place to start</p>
<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>diamond <span class="nb">help</span>
</code></pre></div></div>

<div class="output highlighter-rouge"><div class="highlight"><pre class="highlight"><code>mostrowski@jen:~$ diamond help
diamond v0.9.21.122 | by Benjamin Buchfink &lt;buchfink@gmail.com&gt;
Licensed under the GNU AGPL &lt;https://www.gnu.org/licenses/agpl.txt&gt;
Check http://github.com/bbuchfink/diamond for updates.

Syntax: diamond COMMAND [OPTIONS]

Commands:
makedb	Build DIAMOND database from a FASTA file
blastp	Align amino acid query sequences against a protein reference database
blastx	Align DNA query sequences against a protein reference database
view	View DIAMOND alignment archive (DAA) formatted file
help	Produce help message
version	Display version information
getseq	Retrieve sequences from a DIAMOND database file
dbinfo	Print information about a DIAMOND database file

General options:
--threads (-p)         number of CPU threads
--db (-d)              database file
--out (-o)             output file
--outfmt (-f)          output format
	0   = BLAST pairwise
	5   = BLAST XML
	6   = BLAST tabular
	100 = DIAMOND alignment archive (DAA)
	101 = SAM

	Value 6 may be followed by a space-separated list of these keywords:

	qseqid means Query Seq - id
	qlen means Query sequence length
	sseqid means Subject Seq - id
	sallseqid means All subject Seq - id(s), separated by a ';'
	slen means Subject sequence length
	qstart means Start of alignment in query
	qend means End of alignment in query
	sstart means Start of alignment in subject
	send means End of alignment in subject
	qseq means Aligned part of query sequence
	sseq means Aligned part of subject sequence
	evalue means Expect value
	bitscore means Bit score
	score means Raw score
	length means Alignment length
	pident means Percentage of identical matches
	nident means Number of identical matches
	mismatch means Number of mismatches
	positive means Number of positive - scoring matches
	gapopen means Number of gap openings
	gaps means Total number of gaps
	ppos means Percentage of positive - scoring matches
	qframe means Query frame
	btop means Blast traceback operations(BTOP)
	staxids means unique Subject Taxonomy ID(s), separated by a ';' (in numerical order)
	stitle means Subject Title
	salltitles means All Subject Title(s), separated by a '&lt;&gt;'
	qcovhsp means Query Coverage Per HSP
	qtitle means Query title

	Default: qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore
--verbose (-v)         verbose console output
--log                  enable debug log
--quiet                disable console output

Makedb options:
--in                   input reference file in FASTA format

... truncated
</code></pre></div></div>

<p>And here is an example command for a diamond blastp search of the protein sequences in bin 10 against the NCBI nr database (from October 2018)</p>
<div class="bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>diamond blastp -q bin.10.prokka.out.faa -d /disks/jen/scratch-ssd/db/nr_oct2018.dmnd -p 10 -f 6 -o bin.10.prokka.dvnr.out
</code></pre></div></div>

<p>If you have many bins it is time to start thinking about an efficient way to process them all. One approach is to create a batch file to run a diamond blastp on every bin. The batch file will utilise a loop to create the list of commands. We will save them in a file called <code class="highlighter-rouge">do-diamond.sh</code></p>

<div class="output highlighter-rouge"><div class="highlight"><pre class="highlight"><code>for i in *faa; do echo "diamond blastp -q $i -d /disks/jen/scratch-ssd/db/nr_oct2018.dmnd -p 10 -f 6 -o $i.dvnr.out"; done &gt; do-diamond.sh
</code></pre></div></div>

<p>check that the contents look sane</p>
<div class="bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>less do-diamond.sh
</code></pre></div></div>

<p>suggest running one of these commands to make sure everything executes correctly</p>

<p>if it is OK then make the file executable (i.e. a shell script) with the following</p>

<p>we need to change the permissions to make this file execuatble</p>
<div class="bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>chmod a+x do-diamond.sh
</code></pre></div></div>

<p>#then run the batch, and wait</p>

<div class="bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>./do-diamond.sh
</code></pre></div></div>

<p>Next we’ll extract taxonomic associations from the accession numbers of the diamond blastp output using BASTA</p>

<div class="bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>basta sequence -b BEST_HIT -d /disks/jen/scratch-ssd/db/.basta/taxonomy/ ju.bin14.dvnr.out ju.bin14.basta.out prot
</code></pre></div></div>

<p>Have a look at the results using bash tools</p>

<div class="bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>

</code></pre></div></div>

<div class="output highlighter-rouge"><div class="highlight"><pre class="highlight"><code>

</code></pre></div></div>

<p>then send the results to krona plot</p>

<div class="bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>cat ju.bin14.basta.out | cut -f 1,3 &gt; ju.bin14.basta.bh.out

/usr/local/anaconda2/bin/basta2krona.py  krona.test.html
</code></pre></div></div>

<blockquote class="challenge">
  <h2 id="questions">Questions:</h2>

  <p>Write a loop to process all of the diamond blastp output files</p>

  <blockquote class="solution">

    <p>for i in *.prokka.out.basta.out;
do
name=$(basename $i .prokka.out.basta.out)
echo “cat $i | cut -f 1,3 &gt; nd.bin.$name.basta.bh.out
done &gt; do-bastatrim.sh</p>
  </blockquote>
</blockquote>

<div class="bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>for i in *.prokka.out.basta.out;
do
name=$(basename $i .prokka.out.basta.out)
echo "cat $i | cut -f 1,3 &gt; nd.bin.$name.basta.bh.out
done &gt; do-bastatrim.sh
</code></pre></div></div>

<div class="bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>/usr/local/anaconda2/bin/basta2krona.py nd.bin10.basta.bh.out,nd.bin12.basta.bh.out,nd.bin.bin.11.basta.bh.out,nd.bin.bin.13.basta.bh.out,nd.bin.bin.15.basta.bh.out,nd.bin.bin.1.basta.bh.out,nd.bin.bin.3.basta.bh.out,nd.bin11.basta.bh.out,nd.bin.bin.10.basta.bh.out,nd.bin.bin.12.basta.bh.out,nd.bin.bin.14.basta.bh.out,nd.bin.bin.16.basta.bh.out,nd.bin.bin.2.basta.bh.out,nd.bin.bin.4.basta.bh.out  krona.test.html
</code></pre></div></div>

<blockquote class="challenge">
  <h2 id="questions-1">Questions:</h2>

  <p>How does the results of taxonomic assignment sequences vary between the different  methods of assignment, top-hit versus LCA?</p>

</blockquote>

<p>Krona provides an excellent interactive viewer to begin to explore hierarchal data, such as taxonomic lineages and protein functional categories (hint!). Look at the structure of the Super-focus output based on the SEED hierarchy of functional categories.</p>

<p>Here is a composite taxonomic assignemnt of all of the contigs generated by the class this year. <code class="highlighter-rouge">student.contigs.LCA</code> refers to predicted protein sequences assigned using the LCA approach as implemented by BASTA. In contrast <code class="highlighter-rouge">student.contigs.bh</code> was generated with the best-hit results for each protein sequence.</p>

<embed width="1000" height="1000" src="../fig/krona.LCAvsbh.html" />
<p>&lt;/embed&gt;</p>

<blockquote class="challenge">
  <h2 id="questions-2">Questions:</h2>

  <blockquote class="language-bash solution">
    <h2 id="solutions">Solutions</h2>

    <div class="output highlighter-rouge"><div class="highlight"><pre class="highlight"><code>
</code></pre></div>    </div>
  </blockquote>
</blockquote>



<blockquote class="keypoints">
  <h2>Key Points</h2>
  <ul>
    
    <li><p>Loops are very useful for repeating commands over multiple files</p>
</li>
    
  </ul>
</blockquote>

</article>
















<div class="row">
  <div class="col-xs-1">
    <h3 class="text-left">
      
      <a href="../09-annotation-of-genome-bins/index.html"><span class="glyphicon glyphicon-menu-left" aria-hidden="true"></span><span class="sr-only">previous episode</span></a>
      
    </h3>
  </div>
  <div class="col-xs-10">
    
  </div>
  <div class="col-xs-1">
    <h3 class="text-right">
      
      <a href="../"><span class="glyphicon glyphicon-menu-up" aria-hidden="true"></span><span class="sr-only">lesson home</span></a>
      
    </h3>
  </div>
</div>


      
      
<footer>
  <div class="row">
    <div class="col-md-6 copyright" align="left">

    </div>
    <div class="col-md-6 help-links" align="right">
	
	
	<a href="/edit/gh-pages/_episodes_rmd/12-taxonomic-assignment.Rmd">Edit on GitHub</a>
	
	
	/
	<a href="/blob/gh-pages/CONTRIBUTING.md">Contributing</a>
	/
	<a href="/">Source</a>
	/
	<a href="/blob/gh-pages/CITATION">Cite</a>
	/
	<a href="mailto:martin.ostrowski@mq.edu.au">Contact</a>
    </div>
  </div>
  <div class="row">
    <div class="col-md-12" align="center">
      Using <a href="https://github.com/carpentries/styles/">The Carpentries style</a>
      version <a href="https://github.com/carpentries/styles/releases/tag/v9.5.3">9.5.3</a>.
    </div>
  </div>
</footer>

      
    </div>
    
<script src="../assets/js/jquery.min.js"></script>
<script src="../assets/js/bootstrap.min.js"></script>
<script src="../assets/js/lesson.js"></script>
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
  ga('create', 'UA-37305346-2', 'auto');
  ga('send', 'pageview');
</script>

  </body>
</html>
